src/api/app/models/kiwi/image.rb

```diff
@@ -54,7 +54,7 @@ def self.build_from_xml(xml_string, md5)
 
       repositories.reject{ |repository| repository['source']['path'] == 'obsrepositories:/' }.each.with_index(1) do |repository, index|
         attributes = {
-          repo_type:   repository['type'],
+          repo_type:   repository['type'].in?(%w[yast2 rpm-dir]) ? 'rpm-md' : repository['type'],
```

Comment:
Please drop the ``[]`` if you use ``%w``

Reply:
I would like to use the ``REPO_TYPES`` constant here instead of writing it again here. Also with this condition I could ass ``foobar`` as type which would also be wrong so I would prefer to check for valid repositories (white list vs black list)

```
repository['type'].in?(Kiwi::Repository::REPO_TYPES) ? repository['type'] : 'rpm-md' ,
```

Also maybe you could move this to a dedicated function as it is already a little bit hard to understand. For instance

```
def repository_type(repository)
  repository['type'].in?(%w[yast2 rpm-dir]) ? 'rpm-md' : repository['type']
end
```

Reply:
> repository['type'].in?(Kiwi::Repository::REPO_TYPES) ? repository['type'] : 'rpm-md'

@ChrisBr , but in this case, you are allowing any repo_type including the empty type and we don't want that.

I like moving it into a method, thanks :smile_cat: